commit/tokenprice/outcome.go

```diff
@@ -0,0 +1,169 @@
+		p.lggr,
+		"FeedTokenPrices",
+		aggObs.FeedTokenPrices,
+		shared.TwoFPlus1(fFeedChain),
+		func(vals []cciptypes.TokenPrice) cciptypes.TokenPrice {
+			return shared.Median(vals, shared.TokenPriceComparator)
+		},
+	)
+
+	feeQuoterUpdatesConsensus := shared.GetConsensusMapAggregator(
+		p.lggr,
+		"FeeQuoterUpdates",
+		aggObs.FeeQuoterTokenUpdates,
+		shared.TwoFPlus1(fDestChain),
+		feeQuoterAggregator,
+	)
+
+	consensusObs := ConsensusObservation{
+		FChain:                fChains,
+		Timestamp:             shared.Median(aggObs.Timestamps, shared.TimestampComparator),
+		FeedTokenPrices:       feedPricesConsensus,
+		FeeQuoterTokenUpdates: feeQuoterUpdatesConsensus,
+	}
+
+	return consensusObs, nil
+}
+
+// feeQuoterAggregator aggregates the fee quoter updates by taking the median of the prices and timestamps
+var feeQuoterAggregator = func(updates []shared.TimestampedBig) shared.TimestampedBig {
+	timestamps := make([]time.Time, len(updates))
+	prices := make([]cciptypes.BigInt, len(updates))
+	for i := range updates {
+		timestamps[i] = updates[i].Timestamp
+		prices[i] = updates[i].Value
+	}
+	medianPrice := shared.Median(prices, shared.BigIntComparator)
+	medianTimestamp := shared.Median(timestamps, shared.TimestampComparator)
+	return shared.TimestampedBig{
+		Value:     medianPrice,
+		Timestamp: medianTimestamp,
+	}
+}
+
+// selectTokensForUpdate checks which tokens need to be updated based on the observed token prices and
+// the fee quoter updates
+// a token is selected for update if it meets one of 2 conditions:
+// 1. if time passed since the last update is greater than the stale threshold
+// 2. if deviation between the fee quoter and feed exceeds token's configured threshold
+func (p *processor) selectTokensForUpdate(
+	obs ConsensusObservation,
+) []cciptypes.TokenPrice {
+	var tokenPrices []cciptypes.TokenPrice
+	cfg := p.cfg.OffchainConfig
+	tokenInfo := cfg.TokenInfo
+
+	for token, feedPrice := range obs.FeedTokenPrices {
+		lastUpdate, exists := obs.FeeQuoterTokenUpdates[token]
+		if !exists {
+			// if the token is not in the fee quoter updates, then we should update it
+			tokenPrices = append(tokenPrices, cciptypes.TokenPrice{
+				TokenID: token,
+				Price:   cciptypes.NewBigInt(feedPrice.Price.Int),
+			})
+			continue
+		}
+
+		ti, ok := tokenInfo[token]
+		if !ok {
+			p.lggr.Warnf("could not find token info for token %s", token)
+			continue
+		}
+
+		nextUpdateTime := lastUpdate.Timestamp.Add(cfg.TokenPriceBatchWriteFrequency.Duration())
+		if obs.Timestamp.After(nextUpdateTime) {
+			tokenPrices = append(tokenPrices, cciptypes.TokenPrice{
+				TokenID: token,
+				Price:   cciptypes.NewBigInt(feedPrice.Price.Int),
+			})
+		} else if shared.Deviates(feedPrice.Price.Int, lastUpdate.Value.Int, ti.DeviationPPB.Int64()) {
+			tokenPrices = append(tokenPrices, cciptypes.TokenPrice{
+				TokenID: token,
+				Price:   cciptypes.NewBigInt(feedPrice.Price.Int),
+			})
+		}
```

Comment:
nit: make it clear that both cases do the same thing
```suggestion
		shouldUpdate := obs.Timestamp.After(nextUpdateTime) || shared.Deviates(feedPrice.Price.Int, lastUpdate.Value.Int, ti.DeviationPPB.Int64())
		if shouldUpdate {
			tokenPrices = append(tokenPrices, cciptypes.TokenPrice{
				TokenID: token,
				Price:   cciptypes.NewBigInt(feedPrice.Price.Int),
			})
		}
```